src/Domain/BlockLogicalCoordinates.hpp

```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <ostream>
+#include <vector>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
+template <size_t VolumeDim, typename TargetFrame>
+class Domain;
+namespace PUP {
+class er;
+}  // namespace PUP
+
+/// Holds block logical coords for a single point.
+template <size_t Dim>
+struct BlockLogicalCoordHolder {
```

Comment:
I think this should be replaced by `IdPair<BlockId, tnsr::I<double, Dim, Frame::Logical>>`, other than that I'm happy with it.

Reply:
IdPair is not yet implemented. So should I wait for #874 or should this be merged as is with the intent of changing to IdPair later?

Reply:
I would prefer to wait until #784 is merged. I think @wthrowe already looked at the previous incarnation of it. I'll ping @kidder to take a look at it as well.